video/mp_image.c

```diff
@@ -549,6 +549,15 @@ void mp_image_copy_attributes(struct mp_image *dst, struct mp_image *src)
     assign_bufref(&dst->dovi_buf, src->dovi_buf);
     assign_bufref(&dst->film_grain, src->film_grain);
     assign_bufref(&dst->a53_cc, src->a53_cc);
+
+    for (int n = 0; n < dst->num_ff_side_data; n++)
+        av_buffer_unref(&dst->ff_side_data[n].buf);
+    MP_RESIZE_ARRAY(NULL, dst->ff_side_data, src->num_ff_side_data);
+    dst->num_ff_side_data = src->num_ff_side_data;
+    for (int n = 0; n < dst->num_ff_side_data; n++) {
+        dst->ff_side_data[n].type = src->ff_side_data[n].type;
+        dst->ff_side_data[n].buf = av_buffer_ref(src->ff_side_data[n].buf);
```

Comment:
> If we want to follow the previously utilized style of OOM-handling, MP_HANDLE_OOM(dst->ff_side_data[n].buf), maybe?

Sure.

> And since the specific side data entry would then be getting poked for the 3rd time, possibly introducing struct mp_ff_side_data *dst_sd (and possibly same for the src?) would make sense?

It fits nicely in 80 columns. I find it more readable this way, to make it symmetrical I would also add `src_sd` and it then a it is 312 characters vs 247 of current loop. And imho makes it more verbose than is can be.